lib/web_git/diff.rb

```diff
@@ -0,0 +1,233 @@
+          log.split.first
+        end
+      end
+
+      def self.file_diffs(diff)
+        diff.scan(/diff --git.*?(?=diff --git|\z)/m)
+      end
+
+      def self.match_other_files(line, file, filenames)
+        filenames.each do |other_file|
+          if file != other_file
+            # It looks like:
+            #   --- a/<path-to-file>
+            #   +++ b/<path-to-file>
+            if line.include?('diff --git a/' + other_file + ' b/' + other_file)
+              return true
+            end
+          end
+        end
+        false
+      end
+
+      def self.get_file_names(commit)
+        Dir.chdir(Rails.root) do
+          if commit.blank?
+            filenames = `git diff --name-only`
+          else
+            filenames = `git diff-tree --no-commit-id --name-only -r #{commit}`
+          end
+          filenames.split("\n")
+        end
+      end
+
+      def self.get_last_diff
+        Dir.chdir(Rails.root) do
+          `git diff -M HEAD~1`
+        end
+      end
+
+      def self.get_last_left(diff)
+        filenames = get_file_names(get_last_commit_hash)
+        files = file_diffs(diff)
+        ones = {}
+        files.each_with_index do |file, i|
+          file_content = ""
+          lines = file.split("\n").drop(4)
+          lines.each do |line|
+            # The line numbers in the output of a git diff match this regex
+            # @@ -61,18 +61,15 @@
+            if !line.match?(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/)
+              if line.first != "+"
+                line.slice!(0)
+                file_content += line + "\n"
+              end
+            else
+              file_content += "\n"
+            end
+          end
+          ones[filenames[i]] = file_content.chomp "\n"
+        end
+        ones
+      end
+
+      def self.get_last_right(diff)
+        filenames = get_file_names(get_last_commit_hash)
+        files = file_diffs(diff)
+        ones = {}
+        files.each_with_index do |file, i|
+          file_content = ""
+          lines = file.split("\n").drop(4)
+          lines.each do |line|
+            if !line.match?(/@@ ([-]\d+,\d+\s[+]\d+,\d+) @@/)
+              if line.first != "+"
+              elsif line.first == "+"
+                line.slice!(0)
+                file_content += line + "\n"
+              end
+            else
+              file_content += "\n"
+            end
+          end
+          ones[filenames[i]] = file_content.chomp "\n"
+        end
+        ones
+      end
+
+      def self.last_to_html(diff)
+        left_hash = get_last_left(diff)
+        right_hash = get_last_right(diff)
+
+        html_output = '<link rel="stylesheet"' +
+          'href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/' +
+          'bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/' +
+          '1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">'
+        html_output += '<style>'
+        html_output += Diffy::CSS
```

Comment:
I'm a little surprised that it appears the font in the diff is just Times New Roman, or some non-monospace font. We should specify some monospace font, but why isn't Diffy:CSS doing it already?

Reply:
@raghubetina The last commit is still rendering in an iframe. I didn't change that so it's not showing in the diff. Does it make more sense to have this out of an iframe too?

Reply:
Oh I see now. The reason I was using an iframe before is because the CLI would output an entire HTML document (with its own <head>, etc) — now that we have more fine grained control (I think?) in generating the HTML, I think we can drop the iframe and just build our own single HTML document. What do you think?

Reply:
Definitely! I think it makes sense to depart from an iframe.  I didn't want to make too many visual changes without talking to you about it first.

Reply:
Agreed. Let's ship it as-is first and get things running on repl.it; then we can use the foundation you've laid to improve things.